types/vote.go

```diff
@@ -326,6 +326,23 @@ func (vote *Vote) ToProto() *tmproto.Vote {
 	}
 }
 
+func VotesToProto(votes []*Vote) (res []*tmproto.Vote) {
+	if votes == nil {
+		return nil
+	}
+
+	res = make([]*tmproto.Vote, len(votes))
+	for i, vote := range votes {
+		v := vote.ToProto()
+		// protobuf crashes when serializing "repeated" fields with nil elements
```

Comment:
Do we really want to put `nil` here, or should we just skip this element entirely and append only those that are non-nil?

Reply:
I stated that poorly: What I mean is, for the ones that are `nil` on input, should we be synthesizing empties or skipping them?

Reply:
The point of this PR was to rebase what was done last year so we can build on it. If you're OK with it, I'd propose to leave it as it is in this PR, then there is #7655 which is about fixing all these issues from the initial implementation

Reply:
Actually, I re-read my answer above and thought... "don’t put off til tomorrow what you can do today" :-)
So removed the empty votes from the list as suggested

Reply:
Thanks! I'd also have been fine with deferring it for another PR, I just couldn't see an obvious need for the empties to be encoded. 🙂